pkg/controllermanager/controller/seed/seed_lifecycle_reconcile.go

```diff
@@ -127,6 +129,25 @@ func (c *livecycleReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 		}
 	}
 
+	// If the gardenlet's client certificate is expired and the seed belongs to a `ManagedSeed` then we reconcile it in
+	// order to re-bootstrap the gardenlet.
+	if seed.Status.ClientCertificateExpirationTimestamp != nil && seed.Status.ClientCertificateExpirationTimestamp.UTC().Before(time.Now().UTC()) {
+		managedSeed := &seedmanagementv1alpha1.ManagedSeed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      seed.Name,
+				Namespace: v1beta1constants.GardenNamespace,
+			},
+		}
+		if err := c.gardenClient.Client().Get(ctx, client.ObjectKeyFromObject(managedSeed), managedSeed); err != nil && !apierrors.IsNotFound(err) {
+			return reconcileResult(err)
+		} else if err == nil {
```

Comment:
Perhaps this code could be extracted to an utility funciton, e.g. `kutil.GetManagedSeedByName` that would return nil if the managed seed doesn't exist? I think this is one of few places where such logic exists.

Reply:
Yes, in the `handleManagedSeedIfSeedBelongsToIt` function that you added.